modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -140,10 +140,21 @@ export class Transaction extends BaseTransaction {
     if (!this._solTransaction) {
       throw new ParseTransactionError('Empty transaction');
     }
+
+    let durableNonce: DurableNonceParams | undefined;
+    if (this._solTransaction.nonceInfo) {
+      const nonceInstruction = SystemInstruction.decodeNonceAdvance(this._solTransaction.nonceInfo.nonceInstruction);
+      durableNonce = {
+        walletNonceAddress: nonceInstruction.noncePubkey.toString(),
+        authWalletAddress: nonceInstruction.authorizedPubkey.toString(),
+      };
+    }
+
     const result: TxData = {
       id: this.id,
       feePayer: this._solTransaction.feePayer?.toString(),
       nonce: this.getNonce(),
+      durableNonce: durableNonce,
```

Comment:
So here in TxData durable nonce could be undefined?
We should build a TxData with and without this field?
@CapnMigraine , @joey-bai what you recommend here?

Reply:
Undefined should be fine. Idk whats gonna be this field data be used for

Reply:
@rliberal-simtlix yes, the blockhash/nonce can come from two sources:
- recent value from full node
- value from a durable nonce account

`durableNonce` is used to indicate that the nonce value comes from the nonce account.
if this field is empty, then the nonce is a recent value from the full node.

Reply:
Sounds good. My question was related in order to add increase the payload size of tx, which base on the fee the tx per byte could increase cost. But in that case that you mention i think is ok.
Thanks for the clarification.